contrib/create-ledgers/src/main.rs

```diff
@@ -0,0 +1,99 @@
+use {
+    solana_sdk::{
+        signature::{Keypair, read_keypair_file, Signer,},
+        commitment_config::CommitmentConfig,
+    },
+    solana_client::{
+        rpc_client::RpcClient,
+    },
+};
+
+mod programs;
+mod utils;
+
+fn example_ledger(client: &RpcClient, payer: &Keypair, program_data: &Vec<u8>, account_data: &Vec<u8>) {
+    // Set Up Buffer Accounts
+    let buffer_account_deploy = programs::set_up_buffer_account_transactions(&client, &payer, &program_data);
+    let buffer_account_upgrade = programs::set_up_buffer_account_transactions(&client, &payer, &program_data);
+    let buffer_account_redeploy = programs::set_up_buffer_account_transactions(&client, &payer, &program_data);
+
+    // Deploy Program
+    let (program_account, deploy_program_instructions) = programs::deploy_program_instructions(&client, &payer, None, &buffer_account_deploy, program_data.len());
+    let transaction = utils::create_message_and_sign(&deploy_program_instructions, &payer, vec![&payer, &program_account], client.get_latest_blockhash().unwrap());
+    let _ = client.send_and_confirm_transaction(&transaction);
```

Comment:
We should fail if any of these calls fail.

Reply:
We want some transactions to fail though

Reply:
sure! for cases where we expect them to succeed, we should surface the error not fail silently and continue